### PR TITLE
Tdr 508 save parent folder name

### PR DESCRIPTION
**Pull request by Lholbourn**
These changes add functions to add `ParentFolder` name to the Consignment table. This is done at the same stage in which we save files to the `File` table in order to keep the `File` and `Consignment` table in sync. 

The changes include associated tests and a slight change to the test DB to add `ChecksumMatches` due to PR https://github.com/nationalarchives/tdr-consignment-api/pull/76  not being merged yet. May have to coordinate on which PR should be merged first.

### CODE REVIEW

**Comment by suzannehamilton** [comment]
Looks like you have a test failure on Jenkins.

```
14:48:33  [info] FileServiceSpec:
14:48:33  [info] createFile
14:48:33  [info] - should create a file given correct arguments *** FAILED ***
14:48:33  [info]   org.mockito.exceptions.verification.SmartNullPointerException: You have a NullPointerException here:
14:48:33  [info] -> at uk.gov.nationalarchives.tdr.api.service.FileService.addFile(FileService.scala:25)
```

**Comment by Lholbourn** [comment]
> Looks like you have a test failure on Jenkins.
> 
> ```
> 14:48:33  [info] FileServiceSpec:
> 14:48:33  [info] createFile
> 14:48:33  [info] - should create a file given correct arguments *** FAILED ***
> 14:48:33  [info]   org.mockito.exceptions.verification.SmartNullPointerException: You have a NullPointerException here:
> 14:48:33  [info] -> at uk.gov.nationalarchives.tdr.api.service.FileService.addFile(FileService.scala:25)
> ```

I managed to fix this and a few ScalaStyle issues, should all be good now :+1:

File: src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/ConsignmentRepositorySpec.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+class ConsignmentRepositorySpec extends AnyFlatSpec with ScalaFutures with Matchers {
+  implicit val executionContext: ExecutionContext = ExecutionContext.Implicits.global
+
+  "addParentFolder" should "add parent folder name to an existing consignment row" in {
+    val db = DbConnection.db
+    val consignmentRepository = new ConsignmentRepository(db)
+    val consignmentId = UUID.fromString("0292019d-d112-465b-b31e-72dfb4d1254d")
+
+    TestUtils.createConsignment(consignmentId, userId)
+    TestUtils.createFile(UUID.fromString("813231b3-2fbd-4869-abdf-0c42624c4d57"), consignmentId)
```
Comment (suzannehamilton) [inline src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/ConsignmentRepositorySpec.scala:26]: Are these files necessary for this test? At a high level, we know that adding files happens at the same time as adding a folder name, but at a low level, the `addParentFolder` isn't related to the files at all.

**Review by suzannehamilton — COMMENTED** [review]
Looks good - just one comment about some test data.

File: src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/ConsignmentRepositorySpec.scala (hunk 1 above)
Comment (Lholbourn) [reply src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/ConsignmentRepositorySpec.scala:26]: I hadn't thought of that actually! I've removed it and the test functions the same so I think it's best to just take them out since it doesn't really serve any purpose :+1:

**Review by suzannehamilton — APPROVED** [review]
Looks good, thanks! Is this OK to merge, or were you coordinating that with Sam because there's some overlap with #76?

**Comment by Lholbourn** [comment]
> Looks good, thanks! Is this OK to merge, or were you coordinating that with Sam because there's some overlap with #76?

Awesome, thanks! I'll give him a shout to check what would be best :smile:
